index.bs

```diff
@@ -6064,6 +6076,205 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+
+If the [=client device=] is incapable of generating a [=hardware-bound device key pair=], or the registration or authentication operation fails for any reason, this extension is ignored and no [=hardware-bound device key pair=] is created. In this case, there is no [=devicePubKey=] extension output generated.
+
+The [=hardware-bound device key pair=] is not on its own a [=user credential=] and does not have its own [=credential ID=]. Instead, the returned [=device public key=] is a device-specific contextual attribute of its associated [=user credential=]. That is, when that [=user credential=] is used&mdash;along with the [=devicePubKey=] extension&mdash;on a particular [=client device=], a particular [=device public key=] is returned by the extension.
+
+Issue(1665): TODO: explicitly add the notion of "synced user credentials" to the spec.
+
+
+### Relying Party Usage ### {#sctn-device-publickey-extension-usage}
+
+This extension is intended for use by those [=[RPS]=] employing risk-analysis systems informing their sign-in decisions. This extension provides a "device continuity" signal when used consistently with both {{CredentialsContainer/create()|navigator.credentials.create()}} and {{CredentialsContainer/get()|navigator.credentials.get()}} operations:
+
+A new [=device public key=] is returned to the [=[RP]=] (along with a new [=credential=]) when the `devicePubKey` extension is used with a {{CredentialsContainer/create()|navigator.credentials.create()}} call. The same [=device public key=] is returned to the [=[RP]=] along with [=assertions=] generated using the latter credential&mdash;i.e., via {{CredentialsContainer/get()|navigator.credentials.get()}} calls&mdash;if those {{CredentialsContainer/get()|get()}} calls occur on the <i>same device</i> as the {{CredentialsContainer/create()|create()}} call that created the credential.
+
+However, when a user's credential is synced to a different device, the [=[RP]=] will receive a new [=device public key=] on the initial {{CredentialsContainer/get()|navigator.credentials.get()}} invocation on the other device. Note that a synced credential will be exercised for authentication on a different device without a {{CredentialsContainer/create()|navigator.credentials.create()}} having been performed on that new device.
```

Comment:
In this situation, the credential that is used for the assertion will be the same, so an RP that does NOT check or look for DPK will still be able to validate the user's public key as before? Or in this situation, will this provide a new public key for the user, and if an RP does not check DPK then will they assume it's a bad assertion? IE can an RP freely ignore DPK if they wish.

Reply:
> In this situation, the credential that is used for the assertion will be the same, so an RP that does NOT check or look for DPK will still be able to validate the user's public key as before? [...] IE can an RP freely ignore DPK if they wish.

Correct, the use case is for when a credential private key is available on multiple devices. An RP that doesn't care whether private keys are hardware-confined or cloud-synced can safely ignore the extension and won't notice a difference between signatures made with the same credential private key on different devices.

> Or in this situation, will this provide a new public key for the user

It will, but it's a separate public key that is only used for the signature in the `devicePubKey` extension output. It does not replace the existing (synced) credential private key, it only adds an additional signature in the assertion response.

Reply:
@emlun had a couple questions for my understanding! So the idea here is: the synced credential keypair for e.g. Apple's Passkey would be the one actually used for authentication, and therefore is "synced" across relevant devices?

Similarly, the `userCredentialId` (referred to as the [credentialId](https://www.w3.org/TR/webauthn/#credential-id) in the spec?) would be usable across devices to fetch the synced credential transparently, as if it's a roaming security key (but really the interaction will be with the actual platform device), right?*

Also, so this means the DPK extension pattern basically exists to help RPs understand and validate the [existing/new] devices associated with the main synced credential, after which point that `dpk` is not actually used for authentication?

*This might be a deeper discussion or out of scope on this PR, but presumably this would show up and be filterable/selectable as a platform authenticator, even though the underlying credential is synced, right? What happens today in cases such as phone-as-security-key -- is it filtered as a roaming authenticator?

Thanks so much for your time!

Reply:
@mpeng-okta 

> Also, so this means the DPK extension pattern basically exists to help RPs understand and validate the [existing/new] devices associated with the main synced credential, after which point that dpk is not actually used for authentication?

Yes, the DPK is designed to be a device/context signal for relying parties to detect that the multi-device credential is being presented from the same or a different device/context. The DPK is never backed up and can only live in the loacl authenticator.

The primary use case would be in high value consumer or enterprise-like use cases where you want to offer the good UX of a phone + multi-device credential, but you need additional assurance. So when you detect a DPK for the first, you may require step up via an OOB mechanism to challenge the user. Then each time you see that exact multi-device credential + DPK pair again, you can just proceed without step up.

Reply:
Gotcha, that makes sense. And by never "backed up", it means that it's never conveyed to the central synced credential/keychain/etc right -- RPs should still persist DPK/aaguid/signature/multi-device credential mapping as mentioned, so they can do a quick byte-wise comparison next time.

Reply:
Ah I see, thanks! I'll check out the fido-dev group for questions not immediately pertaining to this spec yeah. And the developer docs are essentially that of the spec, i.e. https://www.w3.org/TR/webauthn/ right? (I'm aware of other WebAuthn dev docs but they're generally specific to the company AFAIK)

Reply:
We are working on developer-centric docs that focus on implementing multi-device credentials.